### PR TITLE
[Merged by Bors] -  Support monitor selection for all window modes.

### LINKED ISSUE #5875: winit ignores monitor selection in fullscreen
## Bevy version

0.8.1

## \[Optional\] Relevant system information

Windows 10

## What you did

```rust
app.insert_resource(WindowDescriptor {
            mode: WindowMode::BorderlessFullscreen,
            position: WindowPosition::Centered(MonitorSelection::Number(42)),
            ..default()
        });
```

## What went wrong

Primary monitor is always selected, no error whatever the monitor number asked (which obviously is nonsense in this example).

## Additional information

Primary monitor is hard-coded here:

https://github.com/bevyengine/bevy/blob/7a29c707bf6c79b3dab6e79048818836cdc26bbe/crates/bevy_winit/src/winit_windows.rs#L29

`WindowDescriptor::position` (which contains the monitor ID) is only ever used when _not_ in fullscreen.

### PR DESCRIPTION
**Pull request by devil-ira**
# Objective
Support monitor selection for all window modes.
Fixes #5875.

## Changelog

* Moved `MonitorSelection` out of `WindowPosition::Centered`, into `WindowDescriptor`.
* `WindowPosition::At` is now relative to the monitor instead of being in 'desktop space'.
* Renamed `MonitorSelection::Number` to `MonitorSelection::Index` for clarity.
* Added `WindowMode` to the prelude.
* `Window::set_position` is now relative to a monitor and takes a `MonitorSelection` as argument.

## Migration Guide

`MonitorSelection` was moved out of `WindowPosition::Centered`, into `WindowDescriptor`.
`MonitorSelection::Number` was renamed to `MonitorSelection::Index`.
```rust
// Before
.insert_resource(WindowDescriptor {
    position: WindowPosition::Centered(MonitorSelection::Number(1)),
    ..default()
})
// After
.insert_resource(WindowDescriptor {
    monitor: MonitorSelection::Index(1),
    position: WindowPosition::Centered,
    ..default()
})
```
`Window::set_position` now takes a `MonitorSelection` as argument.
```rust
window.set_position(MonitorSelection::Current, position);
```

### CODE REVIEW

**Review by afonsolage — APPROVED** [review]

**Review by Weibye — APPROVED** [review]
This is great, thanks!

**Review by Sheepyhead — APPROVED** [review]

File: crates/bevy_window/src/window.rs
Comment (mockersf) [inline crates/bevy_window/src/window.rs:752]: it would be nice to say how to specify the selected monitor

File: crates/bevy_window/src/window.rs
Comment (mockersf) [inline crates/bevy_window/src/window.rs:756]: these two lines feels a lot redundant

**Comment by alice-i-cecile** [comment]
I'm going to wait on the doc improvements and then merge this in :) Let me know if you'd like help.

File: crates/bevy_window/src/window.rs
Comment (devil-ira) [reply crates/bevy_window/src/window.rs:756]: `window's top-left corner` and `top-left corner of the monitor` are not redundant.

**Pull request closed without merging**
